datafusion/core/src/physical_plan/sorts/sort.rs

```diff
@@ -38,13 +38,14 @@ use crate::physical_plan::{
     RecordBatchStream, SendableRecordBatchStream, Statistics,
 };
 use crate::prelude::SessionConfig;
-use arrow::array::{make_array, Array, ArrayRef, MutableArrayData};
+use arrow::array::{make_array, Array, ArrayRef, MutableArrayData, UInt32Array};
 pub use arrow::compute::SortOptions;
 use arrow::compute::{concat, lexsort_to_indices, take, SortColumn, TakeOptions};
 use arrow::datatypes::SchemaRef;
 use arrow::error::ArrowError;
 use arrow::ipc::reader::FileReader;
 use arrow::record_batch::RecordBatch;
+use arrow::row::{OwnedRow, Row, RowConverter, SortField};
```

Comment:
Using OwnedRow is likely to seriously hurt performance, as it will allocate per row.

What do you think about getting the v1 in that doesn't preserve the row encoding, and then we can iterate on v2 in a separate PR. I suspect we will want to preserve rows, and a list of sorted indices.

Reply:
That sound's good. I will just revert this 2nd commit then. Thanks!